.github/workflows/update_sdk_methods.py

```diff
@@ -0,0 +1,842 @@
+    for api in proto_map.keys():
+        api_url = proto_map[api]["url"]
+        api_name = proto_map[api]["name"]
+
+        api_page = urlopen(api_url)
+        api_html = api_page.read().decode("utf-8")
+
+        ## protos are presented in plaintext, so we must match by expected raw text:
+        proto_regex = 'type ' + regex.escape(api_name) + '[^{]*\{([^}]+)\}'
+        search = regex.search(proto_regex, api_html)
+        match_output = search.group()
+        split = match_output.splitlines()
+
+        for line in split:
+            line = line.strip()
+            if line[0].isupper():
+                separator = "("
+                line = line.split(separator, 1)[0]
+                ## Append to proto_map for use later:
+                proto_map[api]["methods"].append(line)
+   
+    return proto_map
+
+## bs4 scraper wrapper used frequently in parse():
+def make_soup(url):
+   try:
+       page = urlopen(url)
+       html = page.read().decode("utf-8")
+       return BeautifulSoup(html, "html.parser")
+   except urllib.error.HTTPError as err:
+       print(f'An HTTPError was thrown: {err.code} {err.reason} for URL: {url}')
+
+## Fetch sdk documentations for each language in sdks array, by language, by type, by resource, by method.
+def parse(type, names):
+
+## TODO:
+## - Unify returned method object form. Currently returning raw method usage for Go, and by-param, by-return (and by-raise)
+##   breakdown for each method for Python and Flutter. Let's chat about which is useful, and which I should throw away.
+##   Raw usage is I think how check_python_methods.py currently does it. Happy to convert Flutter and Py to dump raw usage,
+##   if you don't need the per-param,per-return,per-raise stuff.
+## - Currently manually adding param details for 'extra' and 'timeout' params for Python. There might be more like this,
+##   that need this same manual treatment, that I haven't found yet.
+## - Edge cases (like bad sdk language, or Go App API DNE) just print for now, need to make into errors.
+
+    ## This parent dictionary will contain all dictionaries:
+    ## all_methods[sdk][type][resource]
+    all_methods = {}
+
+    ## Iterate through each sdk (like 'python') in sdks array:
+    for sdk in sdks:
+
+        ## Build empty dict to house methods:
+        if sdk == "go":
+            go_methods = {}
+            go_methods[type] = {}
+        elif sdk == "python":
+            python_methods = {}
+            python_methods[type] = {}
+        elif sdk == "flutter":
+            flutter_methods = {}
+            flutter_methods[type] = {}
+        else:
+            print("unsupported language!")
+
+        ## Determine SDK URL based on resource type:
+        sdk_url = sdk_url_mapping[sdk]
+
+        ## Iterate through each resource (like 'arm') in type (like 'components') array:
+        for resource in names:
+
+            ## Determine URL form for Go depending on type (like 'component'):
+            if sdk == "go":
```

Comment:
would it make things easier to follow if we relegated this logic to language specific files? Then have one parser that imports all three?